File: Backend/src/main/java/de/hft_stuttgart/winf/proj2/sp/backend/dto/ModuleDto.java
package de.hft_stuttgart.winf.proj2.sp.backend.dto;

import de.hft_stuttgart.winf.proj2.sp.backend.util.Column;
import lombok.*;


/**
 * This Class is for defining the a Module and correlating the correct DB-field to it
 */
@AllArgsConstructor
@Getter
@ToString
@EqualsAndHashCode
public class ModuleDto {

    @Column("module_id")
    private Integer module_id;
    @Column("definition")
    private String definition;
    @Column(value="course", isObject = true)
    private CourseDTO course;

    public ModuleDto() {

    }

}

File: Frontend/SPFrontend/src/app/components/models/ModuleDTO.ts
import {CourseDTO} from "./CourseDTO";

export class ModuleDTO {
  module_id: number;
  course: CourseDTO;
  definition: string;

  constructor(module_id, course, definition) {
    this.module_id = module_id;
    this.course = course;
    this.definition = definition;
  }
}

export class CreateModuleDTO {
  course: CourseDTO;
  definition: string;
  user_id: number;

  constructor(course, definition, user_id) {
    this.course = course;
    this.definition = definition;
    this.user_id = user_id;
  }
}
